### ISSUE TITLE
`mypy` should use `List` when complaining about issues with the `List` type, instead of `list`

### ISSUE THREAD

**Issue by hamirmahal** [python/mypy#18530]
**Bug Report**

`mypy` should use `List` when surfacing errors with the `List` type, not `list`.

It's confusing to engineers if `mypy` uses `list` when complaining about a type error because then engineers are **searching the source code for occurrences of `list`, when the problem is actually with the usage of `List`.**

**To Reproduce**

```python
        self.orders: typing.List[typing.List[int, int]] = []
```

**Expected Behavior**

`mypy` should complain with

```
error: "List" expects 1 type argument, but 2 given  [type-arg]
```

**Actual Behavior**

`mypy` actually complains with

```
error: "list" expects 1 type argument, but 2 given  [type-arg]
```

**Your Environment**

- Mypy version used: `mypy==1.10.0`
- Mypy command-line flags: n/a
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: `Python 3.10.12`

**Comment by A5rocks** [comment]
This is a feature request but I will note that:
 - you can get an underline via `--pretty` which should handle any confusion
 - doing this would be inconsistent with other error messages

**Comment by A5rocks** [comment]
Actually, `--pretty` doesn't have as good of a start to end column as I would expect:
```
x.py:5: error: "list" expects 1 type argument, but 2 given  [type-arg]
    x: typing.List[typing.List[int, int]] = []
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```

I reported this seperately as https://github.com/python/mypy/issues/18531.

**Comment by wyattscarpenter** [comment]
I think we could general this request to, mypy should match the name that you use, even if it's a typing.Whatever alias for a builtin that only varies based on capitalization. Which seems reasonable.
